src/utils/check.ts

```diff
@@ -0,0 +1,63 @@
+import isEmpty from 'lodash/isEmpty';
+import isPlainObject from 'lodash/isPlainObject';
+import type { DataType } from '..';
+import { getOperators } from './format';
+// eslint-disable-next-line import/order -- caused by temporarily mixing require with import
+import { Where } from './sequelize-method';
+
+const DataTypes = require('../data-types');
+
+export function isPrimitive(val: any): val is string | number | boolean {
+  const type = typeof val;
+
+  return ['string', 'number', 'boolean'].includes(type);
+}
+
+export function isColString(value: string): boolean {
+  return (
+    typeof value === 'string'
+    && value.startsWith('$')
+    && value.endsWith('$')
+  );
+}
+
+export function canTreatArrayAsAnd(arr: unknown[]): arr is Array<object | Where> {
+  return arr.some(arg => isPlainObject(arg) || arg instanceof Where);
+}
+
+/**
+ * Determine if the default value provided exists and can be described
+ * in a db schema using the DEFAULT directive.
+ *
+ * @param value Any default value.
+ * @returns yes / no.
```

Comment:
Should we go with Boolean here?

Reply:
Same as other comments, I think we can just completely remove this `@returns` since it doesn't provide any information and the return type is extract from ts types